proto-public/go.mod

```diff
@@ -0,0 +1,16 @@
+module github.com/hashicorp/consul/proto-public
+
+go 1.19
+
+require (
+	github.com/golang/protobuf v1.5.0
```

Comment:
Do we need this?

Reply:
I think so because it go pulled in automatically with `go mod tidy` 🤷🏻